download-artifacts/src/index.js

```diff
@@ -0,0 +1,57 @@
+const core = require("@actions/core");
+const github = require("@actions/github");
+const fs = require('fs');
+const path = require('path');
+const extract_zip = require('extract-zip');
+
+async function main() {
+  try {
+    const run_id = parseInt(core.getInput("run-id", {required: true}));
+    // These aren't required but have defaults, so pass in required: true.
+    const pattern = core.getInput("pattern", {required: true});
+    const artifact_path = core.getInput("path", {required: true});
+    const extract = core.getInput("extract", {required: true}) === 'true';
+
+    const dest_path = path.join(github.workspace, artifact_path);
+    fs.mkdirSync(dest_path, {recursive: true});
+
+    const owner = github.context.repo.owner;
+    const repo = github.context.repo.repo;
+
```

Comment:
This is where I'll likely need to extends this to get artifacts from multiple builds if I ever get back to looking at the global report - but it's good for your purposes.

Reply:
This does get artifacts from multiple runs as long as they're attached to the same workflow, I think.

Reply:
I think a workflow_run.id is specific to a particular run of a workflow.   And jobs can't be rerun in a workflow, on a workflow can be rerun.